### PR TITLE
CommonServerPython reference docs

### PR DESCRIPTION
**Pull request by Itay4**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/26399

## Description
 - added docs generator for common server python
 - moved the demisto class doc generation to the python script
 - added processor to address common server python sphinx style
 - added filter processor to skip docstrings

### CODE REVIEW

**Comment by anara123** [comment]
image
For classes we should not print the return type
Also we should not print `__init__`
We should format the enum classes and value in prettier way 
image

**Comment by anara123** [comment]
image
In case of static functions, if we decide to print the function usage then we should do it with the class name, like `FeedIndicatorType.list_all_supported_indicators()`

**Comment by Itay4** [comment]
@glicht @anara123 please review

File: content-repo/create-docs.sh [hunk 1]
```diff
@@ -169,28 +169,20 @@ cp ${CONTENT_GIT_DIR}/Tests/demistomock/demistomock.py .
...
 
-ARTICLES_DIR=${SCRIPT_DIR}/extra-docs/articles
-DEMISTO_CLASS_OVERVIEW="All Python integrations and scripts have available as part of the runtime the \`demisto\` class object. The object exposes a series of API methods which are used to retrieve and send data to the Cortex XSOAR Server.
-
-:::note
-The \`demisto\` class is a low level API. For many operations we provide a simpler and more robust API as part of the  [Common Server Functions](https://xsoar.pan.dev/docs/integrations/code-conventions#common-server-functions).
-:::"
-DEMISTO_CLASS_DOCS_CMD=("./gen_pydocs.py" "-d" "${ARTICLES_DIR}" "-i" "demisto-class" "-t" "'Demisto Class'" "-m" "demisto" "-p" "demisto." "-o" "'${DEMISTO_CLASS_OVERVIEW}'")
-mv demistomock.py demisto.py
+echo "$(tail -n +6 CommonServerPython.py)" > CommonServerPython.py
```
Comment (glicht) [inline content-repo/create-docs.sh:172]: Why are we doing this?

Comment (glicht) [reply content-repo/create-docs.sh:172]: Anyway add a comment explaining this.

**Review by glicht — APPROVED** [review]

File: content-repo/create-docs.sh (hunk 1 above)
Comment (Itay4) [reply content-repo/create-docs.sh:172]: this is to remove the first few lines https://github.com/demisto/content/blob/master/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L1-L5
ill add a comment

**Pull request merged**